src/Controller.ts

```diff
@@ -35,19 +34,31 @@ export const Controller = {
 		return manager;
 	},
 
-	getEnabledEngines: async (): Promise<RuleEngine[]> => {
+	getAllEngines: async (): Promise<RuleEngine[]> => {
 		const engines: RuleEngine[] = [];
-		const logger: Logger = await Logger.child('Controller');
-
 		for (const engine of container.resolveAll<RuleEngine>(Services.RuleEngine)) {
 			await engine.init();
-			if (engine.isEnabled()) {
-				engines.push(engine);
-			} else {
-				logger.trace(`Engine '${engine.getName()}' is disabled}`);
-			}
+			engines.push(engine);
 		}
 
+		if (engines.length == 0) {
+			throw new Error('No engines found');
+		}
+		return engines;
+	},
+
+	getFilteredEngines: async (filteredNames: string[]): Promise<RuleEngine[]> => {
+		const engines: RuleEngine[] = (await Controller.getAllEngines()).filter(e => filteredNames.includes(e.getName()));
+
+		if (engines.length == 0) {
+			throw new Error('No engines found');
```

Comment:
How about including the string[] used for filtering in the error message?
Also, would be good to fetch the error message from a related messages.js file so that we don't have to maintain any user-facing message outside the `.js` files.